source/extensions/transport_sockets/tls/utility.cc

```diff
@@ -152,38 +153,35 @@ int32_t Utility::getDaysUntilExpiration(const X509* cert, TimeSource& time_sourc
   return 0;
 }
 
-absl::optional<std::string> Utility::getX509ExtensionValue(const X509& cert,
-                                                           absl::string_view extension_name) {
-  X509_EXTENSIONS* extensions(X509_get0_extensions(&cert));
-
-  if (extensions == nullptr) {
+absl::optional<std::string>
+Utility::getCertificateExtensionValue(X509& cert, absl::string_view extension_name) {
+  ASN1_OBJECT* oid = OBJ_txt2obj(std::string(extension_name).c_str(), 1 /* don't search names */);
+  if (oid == nullptr) {
     return absl::nullopt;
   }
 
-  const size_t extension_count = sk_X509_EXTENSION_num(extensions);
-
-  for (size_t i = 0; i < extension_count; ++i) {
-    X509_EXTENSION* extension = sk_X509_EXTENSION_value(extensions, i);
-
-    ASN1_OBJECT* extension_object = X509_EXTENSION_get_object(extension);
-    const size_t size = OBJ_obj2txt(nullptr, 0, extension_object, 0);
-    std::vector<char> buffer;
-    // +1 to allow for NULL byte.
-    buffer.resize(size + 1);
-    OBJ_obj2txt(buffer.data(), buffer.size(), extension_object, 0);
+  int pos = X509_get_ext_by_OBJ(&cert, oid, -1);
+  ASN1_OBJECT_free(oid);
```

Comment:
Alternatively, stick `OBJ_txt2obj`'s output in a `bssl::UniquePtr<ASN1_OBJECT>`.

Reply:
Yeah, that's better.